sklearn/ensemble/weight_boosting.py

```diff
@@ -97,22 +105,18 @@ def fit(self, X, y, sample_weight=None):
         if self.learning_rate <= 0:
             raise ValueError("learning_rate must be greater than zero")
 
-        if (self.base_estimator is None or
-                isinstance(self.base_estimator, (BaseDecisionTree,
-                                                 BaseForest))):
-            dtype = DTYPE
-            accept_sparse = 'csc'
-        else:
-            dtype = None
-            accept_sparse = ['csr', 'csc']
-
-        X, y = check_X_y(X, y, accept_sparse=accept_sparse, dtype=dtype,
+        accept_sparse = ['csr', 'csc']
```

Comment:
Why does this need defining separately?

Reply:
Why do we specify sparse formats here but not in _validate_data?

Reply:
Fixed